src/Nethermind/Nethermind.Merge.Plugin/Handlers/ForkchoiceUpdatedHandler.cs

```diff
@@ -235,26 +235,6 @@ public Task<ResultWrapper<ForkchoiceUpdatedV1Result>> Handle(ForkchoiceStateV1 f
                 return ForkchoiceUpdatedV1Result.Error(error, MergeErrorCodes.InvalidPayloadAttributes);
             }
 
-            var spec = _specProvider.GetSpec(newHeadBlock.Number + 1, payloadAttributes.Timestamp);
-
-            if (spec.WithdrawalsEnabled && payloadAttributes.Withdrawals is null)
-            {
-                var error = "PayloadAttributesV2 expected";
-
-                if (_logger.IsInfo) _logger.Warn($"Invalid payload attributes: {error}");
-
-                return ForkchoiceUpdatedV1Result.Error(error, ErrorCodes.InvalidParams);
-            }
-
-            if (!spec.WithdrawalsEnabled && payloadAttributes.Withdrawals is not null)
-            {
-                var error = "PayloadAttributesV1 expected";
-
-                if (_logger.IsInfo) _logger.Warn($"Invalid payload attributes: {error}");
-
-                return ForkchoiceUpdatedV1Result.Error(error, ErrorCodes.InvalidParams);
-            }
-
```

Comment:
Is this fine to do this validation outside the handler? Or should the handler still run even if this would fail?

Reply:
I think fixes with `CompareActivation` might take care of that?